### DISCUSSION TITLE
[5.0] FR: Make 'Add an entry' button in Matrix configurable

### DISCUSSION THREAD

**Discussion by wsydney76** [craftcms/cms#14210] [Ideas] [answered]
Just saw @brandonkelly 's post on colored nested matrix:

![image](https://github.com/craftcms/cms/assets/60010597/4a7dfefb-5580-404f-8f01-3dcc1a2252df)

Looks nice, however, I believe having two buttons with the same caption 'Add an entry' so close to each other will be confusing for editors.

It would be very welcome if the button labeling could be configured/translated, e.g. something like 'Add image to gallery' for the first button. Like the 'Selection Label' in entries fields.

I would even try to avoid the term 'entry' on matrix level this way, although technically correct, but hard to understand for editors.

/W.

**Comment by mmikkel** [comment]
> I would even try to avoid the term 'entry' on matrix level this way, although technically correct, but hard to understand for editors.

Strongly agree.

As a minimum, I think Matrix fields need a field setting similar to other relational fields' "Selection Label" – i.e. making it possible to change the default "Add an entry" to something else (e.g. "Add a block").

However, my personal opinion is that author-facing Craft UI should probably just continue to refer to Matrix entries as "blocks" (or even "block entries", "inline entries", "nested entries" or something like that).

IMO, the fact that Matrix elements are entries under the hood is a technical implementation detail that is completely irrelevant to content authors. Not having a term to differentiate between standalone entries and entries created within Matrix fields is likely going to be confusing. Especially for authors in sites being upgraded to Craft 5, but we've been training authors about "entries" and "blocks" for many years now, and I don't see how conflating the two in the UI - _even if it's technically correct_ – is going to help the AX.

**Comment by brandonkelly** [comment, marked as answer]
Been meaning to improve Matrix button labels, so thanks for the nudge.

As of the next release:

- If only one entry type is selected, the button will be “New [type name]”.
- If the Matrix field is set to the “inline-editable blocks” view mode, the button will be labeled “New block”.
- Otherwise, the button will be labeled “New entry” (which makes more sense for the other view modes, where it’s more obvious you’re creating an entry).

(842ba1586aa5a6f3d632c41186c92dbe7aca8f16)

**Reply by brandonkelly** [reply]
5.0.0-alpha.9 was just tagged with those changes.
